### DISCUSSION TITLE
How to install on a Red Hat based system

### DISCUSSION THREAD

**Discussion by Kitsep** [GameServerManagers/LinuxGSM#3513] [Q&A] [answered]
The instructions for installing ANY of the servers all suggest that you can install on CentOS... but why does the install script include apt-get commands? I have tried to install almost a dozen different servers now, and it appears to be 100% impossible to use Linux GSM on a CentOS machine. I even went to Youtube and followed a start-to-finish tutorial some random dude made for Arma 3. He was able to get his server up and running with no difficulties on Centos 7, but even when I typed out my commands TO THE LETTER identical to his with an absolutely 100% fresh install of CentOS 7, it still fails at the exact same spot as on Cent 8, Fedora WS 32, and Rocky Linux because the script tries to use commands for a Debian based distro

Am I missing something? Has Red Hat support been dropped? FYI I would prefer to use Rocky Linux since Cent 7 and 8 are EOL soon. 

Attached is a screenshot of what happens when I try to run the install script. 

![apt](https://user-images.githubusercontent.com/87346331/125374348-72ebae80-e34c-11eb-82c2-daf1aa629af2.png)

**Comment by dgibbs64** [comment]
I have never seen this error before so I don't know the cause of this issue.

**Comment by rconjoe** [comment, marked as answer]
I'm wondering why info_distro would think you're on debian.

Have you tried installing dependencies manually before running the install script?
For CentOS:
```yum install epel-release
yum install curl wget tar bzip2 gzip unzip python3 binutils bc jq tmux glibc.i686 libstdc++ libstdc++.i686
```
Also, what's the output of `cat /etc/centos-release`?

**Reply by Kitsep** [reply]
I had already manually installed them, but just in case, I verified by running the command you posted, and the result was "nothing to do" as I expected. As for the output, I am tried this on Cent 7, 8 and Rocky, and as I am tired of reinstalling from scratch, I am currently still on Rocky. My output is:

```
[root@***** ~]# cat /etc/rocky-release
Rocky Linux release 8.4 (Green Obsidian)
```
Running in Rocky should be identical to running in CentOS 8, which should be more or less identical to running in CentOS 7. It certainly shouldn't make info_distro think I'm on debian. If I REALLY have to, I can go back to Cent 7 again, but I've wiped and reinstalled so many times now that I would really prefer to avoid that, plus since Cent is EOL soon, I really would like to stay on Rocky. (as an aside, I'll be firing up a Cent 7 server on identical hardware tonight once I get home from work, so perhaps I can do more troubleshooting from Cent 7 once that is ready)

I'm doing this exercise in futility to run an Ark server for my friends. Once I gave up on Linux GSM I ended up just getting a server up and running the manual way (which is another reason why I'm against wiping my Rocky install and going back to Cent 7). I simply wanted to use Linux GSM to simplify the process of running multiple maps simultaneously in a cluster, because doing that manually is a real pain in the rear. 

I want to clarify that even though I have my server up and running, I still want to get Linux GSM working, I don't want to give up completely. Additionally, perhaps this thread may assist someone else with the same issue

**Reply by Kitsep** [reply]
I should also probably clarify that I used the following distros in the following order: I started with Fedora WS 32 since I already had it installed on my server. Then when that failed, I went to Cent 7, which failed in the same spot. Then I tried Cent 8, same failure. After those, I tried Rocky, which is where my first screenshot came from. All of the distros I tried failed in the same spot, in the same way. I thought perhaps I was doing something wrong, but after double and triple checking, and even trying different tutorials from different places using my Rocky install, I can confirm that I am not doing something wrong. So unless there is something specific  about my hardware (which I don't think is the case) then it must be the software. It is also possible that the website may be broken and is just serving the debian version to everyone... I genuinely don't know

**Reply by Kitsep** [reply]
I have an answer, sort of. A friend helped me find the code that detects the distro on github, and if it fails to detect, it seems to default to Debian. It would appear that CentOS 7 and 8 should be automatically detected, but Rocky isn't in the list of distros that the code can recognize. Fedora WS looks like it should be compatible, but it's possible that WS may be the issue there.

Still not sure why Cent 7 and 8 didn't work, as those should be compatible, but at least I have my answer for 2 out of the 4 distros I tried...

**Reply by rconjoe** [reply]
I think this could have been answered simply by looking at the output you sent back to me. 

It is incorrect to assume that running these scripts in Rocky "should be" the same as CentOS. It is not, for the exact reason that the file in /etc/ you looked at for release details is `rocky-release`, not `centos-release`. This is a crucial difference as distro detection in LinuxGSM looks at particular files for distro information, of which rocky is not on the list. This is exactly why I asked, sorry you had so much trouble with it.

I'll make an issue for this tonight if there isn't one made already. I assume Rocky just needs to be treated as if it were RHEL or Cent. Once that is added I would expect you'll have no issues using it in place.

In the meantime, are you able to use an Ubuntu/Debian image? That would solve it until this is fixed.

**Reply by rconjoe** [reply]
ref. https://github.com/GameServerManagers/LinuxGSM/issues/3494

**Reply by Kitsep** [reply]
My assumption was based on other experiences, as I have been using Rocky as a drop-in replacement for Cent 7 (I never truly got onboard with 8) since it was released, and I have had zero incompatibilities. In fact, I've been using Fedora WS 32 and Cent 7 pretty much interchangeably until Rocky came out. Until I tried LinuxGSM (admittedly for my first time) I had never had any issues installing software made for CentOS on Rocky or Fedora WS, using the exact same commands and methods as if I were on CentOS itself. I guess you learn something new about Linux every day, as I had never run into a problem like this, where a piece of software checks the OS name. That is a clever way of making installation simpler, but I suppose this here would be the downside. 

I apologize for the confusion, but I had never encountered anything like this before. I can run Ubuntu or Debian if needed, or I can go back to CentOS 7. I was hoping to jump ship from Cent, but since 7 still has a couple years left before it can be legally killed off, I think that will be my best bet. I'm sure that Rocky will be added to the supported distros WELL before 7 is EOL. Worst case I'll just install Ubuntu server, I use Ubuntu desktop on my main computer anyway so I'm fairly familiar with it.

Thanks!

PS: 

Although this tells us why I had this issue on Fedora WS, Cent 8 and Rocky, I still don't know why I had this issue on Cent 7. I think I have a possibility though. Normally I delete all old partitions whenever I change distros, but since I used the same usernames and passwords between Cent and Fedora after I did the install, it's 100% possible that (assuming I accidentally installed alongside Fedora instead of over the top of it) that the machine booted back into Fedora. So I may have been trying to install LinuxGSM on Fedora when I thought I was installing on CentOS 7. The reason I see this as a possibility is because I probably wouldn't have been able to tell the difference since I was using SSH instead of being at the machine itself. Let's chalk this one up to user error, eh?

**Reply by rconjoe** [reply]
Sure thing. The transition away from CentOS to Rocky/Alma is, at least within this project's scope, a pretty recent development. But the issue is coming up more and more now so I'm sure it will receive due attention :)

**Comment by airtonix** [comment]
Why is this such a huge deal? I mean what is that you guys require that means we cant use Fedora?

**Reply by h3o66** [reply]
You can of course use the ditrubution what you favor, but lgsm does not have any direct support for it.

About the wrong detection of the system:
Usually this is caused by installing `dpkg` on a rpm system
